query-engine/core/src/metrics/registry.rs

```diff
@@ -0,0 +1,307 @@
+use indexmap::IndexMap;
+use metrics::{CounterFn, GaugeFn, HistogramFn, Key};
+use metrics_util::{
+    registry::{GenerationalAtomicStorage, GenerationalStorage, Registry},
+    Histogram as HistogramUtil,
+};
+use parking_lot::RwLock;
+use std::fmt;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::{collections::HashMap, time::Duration};
+
+use tracing::{
```

Comment:
nitpick: no newlines between `use`s and always squash use statements into blocks (i.e. `std::fmt, std::sync => std::{fmt, sync}`